https://invent.kde.org/utilities/ark/-/commit/62d94c62f2fd2052be91dfe565e35a6d43c7d381

From 62d94c62f2fd2052be91dfe565e35a6d43c7d381 Mon Sep 17 00:00:00 2001
From: Elvis Angelaccio <elvis.angelaccio@kde.org>
Date: Sat, 14 Oct 2023 15:21:53 +0200
Subject: [PATCH] Explicitly support new application/vnd.efi.iso mimetype

It has been introduced by the new shared-mime-info 2.3 release. The old
application/x-cd-image is kept as alias (so ark could still open ISO
files just fine), but it broke our mimetype unit test.

While at it, we explicitly add the new entry to the list of supported
mimetypes.
--- a/autotests/kerfuffle/mimetypetest.cpp
+++ b/autotests/kerfuffle/mimetypetest.cpp
@@ -38,7 +38,7 @@ void MimeTypeTest::testMimeTypeDetection_data()
     const QString compressedLzopTarMime = QStringLiteral("application/x-tzo");
     const QString compressedLrzipTarMime = QStringLiteral("application/x-lrzip-compressed-tar");
     const QString compressedLz4TarMime = QStringLiteral("application/x-lz4-compressed-tar");
-    const QString isoMimeType = QStringLiteral("application/x-cd-image");
+    const QString isoMimeType = QMimeDatabase().mimeTypeForFile(QStringLiteral("dummy.iso"), QMimeDatabase::MatchExtension).name();
     const QString debMimeType = QMimeDatabase().mimeTypeForFile(QStringLiteral("dummy.deb"), QMimeDatabase::MatchExtension).name();
     const QString xarMimeType = QStringLiteral("application/x-xar");
     const QString appImageMimeType = QStringLiteral("application/x-iso9660-appimage");
--- a/plugins/libarchive/kerfuffle_libarchive_readonly.json.cmake
+++ b/plugins/libarchive/kerfuffle_libarchive_readonly.json.cmake
@@ -71,6 +71,7 @@
             "application/x-lzop",
             "application/x-source-rpm",
             "application/vnd.debian.binary-package",
+            "application/vnd.efi.iso",
             "application/vnd.ms-cab-compressed",
             "application/x-xar",
             "application/x-iso9660-appimage",
-- 
GitLab
